### ISSUE TITLE
`disallowed_names` for identifiers and not just variables

### ISSUE THREAD

**Issue by ojeda** [rust-lang/rust-clippy#11306]
`disallowed_names` complains about variables, like the lint description explains:

```sh
echo 'fn main() { let foo = 0; }' | rustup run 1.71.1 clippy-driver -Aunused_variables -
```

However, it does not warn on identifiers in general:

```sh
echo 'struct Foo; fn main() {}' | rustup run 1.71.1 clippy-driver -Adead_code -
echo 'fn foo() {} fn main() {}' | rustup run 1.71.1 clippy-driver -Adead_code -
```

This is particularly noticeable in a sample that may use a struct and then instantiate it. I have hit this in a Linux kernel example:

```rust
#![expect(clippy::disallowed_names)]

#[pin_data]
struct Foo {
    #[pin]
    a: Mutex<usize>,
    b: u32,
}

let foo = pin_init!(Foo {
    a <- new_mutex!(42, "Foo::a"),
    b: 24,
});
```

Is it intended that the lint only covers variables? If yes, should there be a `disallowed_idents` even if it overlaps with this one? (`disallowed_types` and others seem to be intended to cover particular paths that should not be used for a given reason, rather than just a name on any path; and even if those lints supported it, then it would still require repeating the name for the different entities).

If it is not intended:
  - Which entities should be cover? All identifiers? Something else?
  - Is it worth renaming it to `disallowed_idents` or similar to be consistent?
  - Should it be case-insensitive (so that the default list `Foo` triggers the lint for e.g. a `struct`)? If not, should the default list be expanded to include `Foo` and so on?

**Comment by CobaltCause** [comment]
FWIW I just noticed that `disallowed_names` also doesn't cover struct field names, which seems odd.
